server/src/main/java/org/elasticsearch/search/lookup/FieldLookup.java

```diff
@@ -37,51 +26,25 @@ MappedFieldType fieldType() {
         return fieldType;
     }
 
-    public Map<String, List<Object>> fields() {
-        return fields;
-    }
-
     /**
      * Sets the post processed values.
      */
-    public void fields(Map<String, List<Object>> fields) {
-        this.fields = fields;
+    public void setValues(List<Object> values) {
+        this.values.clear();
```

Comment:
I wonder if we should throw an error here if valuesLoaded == true. We make sure the callers have clear the object before calling this method?